sql/sql_analyze_stmt.h

```diff
@@ -76,6 +76,8 @@ class Exec_time_tracker
   {
     ulonglong end= measure();
     cycles += end - last_start;
+    if (unlikely(end < last_start))
+      cycles += ULONGLONG_MAX;
```

Comment:
I don't fully understand the intent of this code. Can you please elaborate? This looks to me to be just `cycles--`.

In addition, I don't understand the scenario where `end < last_start` would happen that this would help. `my_timer_cycles()` on Intel I believe this should happen if the computer has been switched on > 100 years, or the execution wasn't pinned to a single CPU / core (in which case, it would be a junk measure anyway).

If it is to fix `my_timer_microseconds()`, the real fix would be to use a monotonic time source (it also wouldn't be a cycles measure, but I'm ignoring that for now :)